pallets/erc20-app/src/lib.rs

```diff
@@ -110,6 +112,10 @@ impl<T: Trait> Module<T> {
 			return Err(Error::<T>::InvalidAssetId.into())
 		}
 
+		if ( T::AccountId::default() == payload.recipient_addr ) {
```

Comment:
why don't allow transfer of erc20 to default address?

Reply:
Default address corresponds to "0x00..00".
The bridge contract (on ethereum) locks and sends in the same step. So a dummy call must inevitable be made when initializing a to token on ethereum for the bridge the way we are for MNG.
To avoid this dummy call from actually doing anything, I call the lock and send function on ethereum using "subNullRecipient", and then these ignore calls in substrate.

Reply:
But we can do the trick much simpler- instead of calling lock and send function on the contract, we can just transfer tokens to the contract using `transfer` function of token contract. As the result, the tokens will be locked on the contract address, but no bridge-related events are triggred

Reply:
For the bridge to work the other way there must be an entry in the ERC20App contract's registry with sufficient amount in favor of the ERC20 token. So when the ERC20 tokens are being locked and sent from ethereum to substrate, the locking of the tokens must be added to the registry with "totalTokens[_tokenAddr] = totalTokens[_tokenAddr].add(_amount);"
This operation is only carried out in the function "sendERC20", which carries out both locking and sending.

Reply:
I was used to think the balance of the contract address serves as a registry. I will doublecheck it and get back to this question

Reply:
See it now. Agree with your solution.